### DISCUSSION TITLE
[Uniques v2] Nested ownership

### DISCUSSION THREAD

**Discussion by jsidorenko** [paritytech/substrate#11800] [Ideas]
There is a nice article about the possible implementation of the nested ownership from the RMRK team [here](https://github.com/rmrk-team/rmrk-substrate/issues/27). It looks like their approach uses a recursion which is a thing we want to avoid within the substrate pallets due to the nested calls limit. I'm wondering if it's possible to come up with an improved implementation that would remove the recursion for nested structures?

**Comment by gilescope** [comment]
If an NFT can own another NFT then it's really an account. That way it could own some fungible tokens also. There's the ED issue which could be solved by putting some dot in the account but maybe there's a more elegant solution?

**Reply by gilescope** [reply]
Is it possible currently to sell / transfer an account to someone else? I guess that might be uncharted territory...  maybe proxies would benefit from the same being able to happen to them?

**Reply by gilescope** [reply]
proxies are more permissioned views on an account so transferring proxies probably doesn't make sense. But transferring sub-accounts probably would.

**Reply by jsidorenko** [reply]
I don't think you can sell your account as it means you'll need to send and "forget" the private key.. Probably even the sub-accounts won't be possible to "sell" since they are derived from your main acc..

Putting some dot for every NFT in order to solve the ED problem is a so-so solution I would say. It won't make sense to use the platform at all in that case. If I have an NFT that would consist of 10 sub-items, I don't wanna put 11 dots for ED, and this is just for one main NFT

**Reply by jsidorenko** [reply]
at the same time if we could find out another way to bypass the ED and make the account tradeable - then yeah, your idea sounds really interesting!

**Reply by jsidorenko** [reply]
I think the idea of creating virtual accounts could be feasible and would allow having the support of the fungible tokens. But it looks more like a bonus feature and not the solution for the lookup/recursion problem we have here..
